### ISSUE TITLE
Unexpected behavior with warm start

### ISSUE THREAD

**Issue by nateagr** [tensorflow/tensorflow#28969]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Centos7
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: Not relevant
- TensorFlow installed from (source or binary): Binary
- TensorFlow version (use command below): v1.12.0-0-ga6d8ffae09 1.12.0
- Python version: 3.6
- Bazel version (if compiling from source): Not relevant
- GCC/Compiler version (if compiling from source): Not relevant
- CUDA/cuDNN version: Not relevant
- GPU model and memory: Not relevant

I've implemented a custom logistic regression which is working as expected. My implementation is based on Estimator API. I managed to train the model with a set of hyperparameter. After  that, I wanted to train another model but with a different learning rate and a warm start using the previously trained model. The learning itself ran succesfully. Since the new learning started from the last checkpoint of the previous learning (due to the warm start), I would expect the new model to start with the variable values of the last checkpoint of the previous learning. But looking at the metrics of tensorboard, It seems the new learning started from scratch. As we can see in the screenshot, the prediction_mean metric starts above 5e-4 and converges around 2.5e-4. The problem here is the same metric prediction_mean already converged around 2.5e-4 for the previous learning. Same observation regarding the metric auc.

![image](https://user-images.githubusercontent.com/41445208/58267669-63800480-7d84-11e9-92db-85d547cf8812.png)

Model trainings are run on yarn and are distributed according to the parameter server strategy.

The following is the piece of code to instantiate the estimator with a warm start:
```
tf.estimator.Estimator(
    model_fn=model_fn_babaflow.model_fn,
    params={
        "optimizer": tf.train.FtrlOptimizer(
            1e-1,
            l2_regularization_strength=1e-1
        )
    },
    model_dir=model_dir,
    warm_start_from="viewfs://path_to_model_dir_of_previous_learning"
)
```

The following is an extract of the log of the chief:

> 2019-05-22 14:59:36,321:INFO:tensorflow: Starting execution chief:0
> 2019-05-22 14:59:36,322:INFO:tensorflow: Broadcasting chief:0/train_eval_start_time = '1558537176.3222742'
> 2019-05-22 14:59:36,323:INFO:tensorflow: Broadcasting chief:0/start = ''
> 2019-05-22 14:59:36,331:INFO:tensorflow: Not using Distribute Coordinator.
> 2019-05-22 14:59:36,912:INFO:tensorflow: Calling model_fn.
> 2019-05-22 14:59:39,613:WARNING:tensorflow: From /tmp/9e20594d-0554-4d16-8b21-122638b68185/install/tensorflow-1.12.0-cp36-cp36m-manylinux1_x86_64.whl.e0ceba8cc1b266d3356296be2708e12fb322668e/tensorflow-1.12.0-cp36-cp36m-manylinux1_x86_64.whl/tensorflow/python/ops/sparse_ops.py:1165: sparse_to_dense (from tensorflow.python.ops.sparse_ops) is deprecated and will be removed in a future version.
> Instructions for updating:
> Create a `tf.sparse.SparseTensor` and use `tf.sparse.to_dense` instead.
> 2019-05-22 14:59:50,953:INFO:tensorflow: Done calling model_fn.
> 2019-05-22 14:59:50,953:INFO:tensorflow: Warm-starting with WarmStartSettings: WarmStartSettings(ckpt_to_initialize_from='viewfs://...', vars_to_warm_start='.*', var_name_to_vocab_info={}, var_name_to_prev_var_name={})
> 2019-05-22 14:59:50,953:INFO:tensorflow: Warm-starting from: ('viewfs://...',)
> 2019-05-22 14:59:50,957:INFO:tensorflow: Warm-starting variable: weights/weights; prev_var_name: Unchanged
> 2019-05-22 15:00:02,025:DEBUG:tensorflow: Initialize variable weights/weights/part_0:0,weights/weights/part_1:0,weights/weights/part_2:0,weights/weights/part_3:0,weights/weights/part_4:0,weights/weights/part_5:0,weights/weights/part_6:0,weights/weights/part_7:0,weights/weights/part_8:0,weights/weights/part_9:0,weights/weights/part_10:0,weights/weights/part_11:0,weights/weights/part_12:0,weights/weights/part_13:0,weights/weights/part_14:0,weights/weights/part_15:0,weights/weights/part_16:0,weights/weights/part_17:0,weights/weights/part_18:0,weights/weights/part_19:0,weights/weights/part_20:0,weights/weights/part_21:0,weights/weights/part_22:0,weights/weights/part_23:0,weights/weights/part_24:0,weights/weights/part_25:0,weights/weights/part_26:0,weights/weights/part_27:0,weights/weights/part_28:0,weights/weights/part_29:0,weights/weights/part_30:0,weights/weights/part_31:0,weights/weights/part_32:0,weights/weights/part_33:0,weights/weights/part_34:0,weights/weights/part_35:0,weights/weights/part_36:0,weights/weights/part_37:0,weights/weights/part_38:0,weights/weights/part_39:0,weights/weights/part_40:0,weights/weights/part_41:0,weights/weights/part_42:0,weights/weights/part_43:0,weights/weights/part_44:0,weights/weights/part_45:0,weights/weights/part_46:0,weights/weights/part_47:0,weights/weights/part_48:0,weights/weights/part_49:0,weights/weights/part_50:0,weights/weights/part_51:0,weights/weights/part_52:0,weights/weights/part_53:0,weights/weights/part_54:0,weights/weights/part_55:0,weights/weights/part_56:0,weights/weights/part_57:0,weights/weights/part_58:0,weights/weights/part_59:0,weights/weights/part_60:0,weights/weights/part_61:0,weights/weights/part_62:0,weights/weights/part_63:0,weights/weights/part_64:0,weights/weights/part_65:0,weights/weights/part_66:0,weights/weights/part_67:0,weights/weights/part_68:0,weights/weights/part_69:0,weights/weights/part_70:0,weights/weights/part_71:0,weights/weights/part_72:0,weights/weights/part_73:0,weights/weights/part_74:0,weights/weights/part_75:0,weights/weights/part_76:0,weights/weights/part_77:0,weights/weights/part_78:0,weights/weights/part_79:0,weights/weights/part_80:0,weights/weights/part_81:0,weights/weights/part_82:0,weights/weights/part_83:0,weights/weights/part_84:0,weights/weights/part_85:0,weights/weights/part_86:0,weights/weights/part_87:0,weights/weights/part_88:0,weights/weights/part_89:0,weights/weights/part_90:0,weights/weights/part_91:0,weights/weights/part_92:0,weights/weights/part_93:0,weights/weights/part_94:0,weights/weights/part_95:0,weights/weights/part_96:0,weights/weights/part_97:0,weights/weights/part_98:0,weights/weights/part_99:0,weights/weights/part_100:0,weights/weights/part_101:0,weights/weights/part_102:0,weights/weights/part_103:0,weights/weights/part_104:0,weights/weights/part_105:0,weights/weights/part_106:0,weights/weights/part_107:0,weights/weights/part_108:0,weights/weights/part_109:0,weights/weights/part_110:0,weights/weights/part_111:0,weights/weights/part_112:0,weights/weights/part_113:0,weights/weights/part_114:0,weights/weights/part_115:0,weights/weights/part_116:0,weights/weights/part_117:0,weights/weights/part_118:0,weights/weights/part_119:0,weights/weights/part_120:0,weights/weights/part_121:0,weights/weights/part_122:0,weights/weights/part_123:0,weights/weights/part_124:0,weights/weights/part_125:0,weights/weights/part_126:0,weights/weights/part_127:0,weights/weights/part_128:0,weights/weights/part_129:0,weights/weights/part_130:0,weights/weights/part_131:0,weights/weights/part_132:0,weights/weights/part_133:0,weights/weights/part_134:0,weights/weights/part_135:0,weights/weights/part_136:0,weights/weights/part_137:0,weights/weights/part_138:0,weights/weights/part_139:0,weights/weights/part_140:0,weights/weights/part_141:0,weights/weights/part_142:0,weights/weights/part_143:0,weights/weights/part_144:0,weights/weights/part_145:0,weights/weights/part_146:0,weights/weights/part_147:0,weights/weights/part_148:0,weights/weights/part_149:0 from checkpoint viewfs://... with weights/weights
> 2019-05-22 15:00:02,026:INFO:tensorflow: Warm-starting variable: bias/bias; prev_var_name: Unchanged
> 2019-05-22 15:00:09,259:DEBUG:tensorflow: Initialize variable bias/bias:0 from checkpoint viewfs://root/... with bias/bias
> 2019-05-22 15:00:09,262:INFO:tensorflow: Create CheckpointSaverHook.
> 2019-05-22 15:00:37,061:INFO:tensorflow: Graph was finalized.

Is there anything wrong about what I'm doing ? Issue already known ?

**Comment by karmel** [comment]
Thanks, @nateagr , for the report. It's hard for us to debug the example as it is, though-- can you provide a minimal code sample that allows us to reproduce? The logs indicate that at least something is being warm-started, but it's hard to tell from that what might be missing.

**Comment by ymodak** [comment]
Automatically closing due to lack of recent activity. Please update the issue when new information becomes available, and we will reopen the issue. Thanks!

**Comment by cocosci** [comment]
If your new estimator contains new Variables that are not contained in the old estimator from the checkpoint, will those new Variables be automatically initialized? Thanks.
